### ISSUE TITLE
Core/GameObject: Restock for chests

### ISSUE THREAD

**Issue by Killyana** [TrinityCore/TrinityCore#23451]
**Description:** The restock for gobs type chest is not implemented or implemented wrongly: https://github.com/TrinityCore/TrinityCore/commit/7011aabb56f061fc889ec42ce32b7605ceb35b53

GAMEOBJECT_TYPE_CHEST = 3
data2: chestRestockTime (time in seconds)
data3: consumable (State: Boolean flag)

Chest that must Restock/despawn:

- The ones with data2=0 AND data3=0: this chests didn't despawn and regenerate loots instantly, they all contains unique quest item
``` sql
SELECT * FROM gameobject_template WHERE TYPE=3 AND data2=0 AND data3=0;
```

- The ones with data2>0 AND data3=0: this chests didn't despawn and regenerate loots after X secs= data2 , they could contains quest item or any kind of item, If the chest contain 2 items or more and only one is looted the restock must reset the looted item inside the gob after X secs= data2
``` sql
SELECT * FROM gameobject_template WHERE TYPE=3 AND data2>0 AND data3=0;
```

- The ones with data2>0 AND data3=1: this chests despawns once empty and regenerate loots after X secs= data2  if not all items are looted (it depawns once empty), they could contains quest items or any kind of item, If the chest contains 2 items or more and only one is looted the restock must reset the looted item inside the gob after X secs= data2
``` sql
SELECT * FROM gameobject_template WHERE TYPE=3 AND data2>0 AND data3=1;
```

- The ones with data2=0 AND data3=1, this chests despawns once empty and doesn't regenerate loots.
``` sql
SELECT * FROM gameobject_template WHERE TYPE=3 AND data2=0 AND data3=1;
```

This must be implemented with https://github.com/TrinityCore/TrinityCore/issues/15730#issuecomment-486457812 to avoid having conflict with respawn/Restock

**Branch(es):** both

**TC rev. hash/commit:** rev. 9083fc149d61

**Comment by Wyrserth** [comment]
BfA retail cannot be used as a reliable source because all the quest-related gameobjects are personal spawns and other players cannot affect it. From sniffs the only thing I could find out is that, after it is looted, the gameobject loses the interactable+sparkle dynamicflags to gain them in the next update tick. I don't yet know if this is just how it worked serverside back then or if it triggers a specific behavior clientside.

**Comment by Wyrserth** [comment]
>The ones with data2=0 AND data3=0: this chests didn't despawn and regenerate loots instantly, they all contains unique quest item

Fixed in 2b1e8d135bd3f3a3ddf7f29471cf6116da223175.

Case two and three are not yet fixed.

**Comment by Wyrserth** [comment]
[In this video](https://youtu.be/rPH9esVAzSs?t=80) the player tries to open a gameobject that is currently restocking, he gets a "There is no loot" error.
